### PR TITLE
Adding user support from ThreadContext

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue #195 *

*Description of changes:*
Adding user support from ThreadContext to integrate with changes in Security Plugin instead of relying on Authorization header (https://github.com/opendistro-for-elasticsearch/security/pull/798). 

Related changes in Common Utils library:
https://github.com/opendistro-for-elasticsearch/common-utils/pull/9

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by skkosuri-amzn — APPROVED** [review]
Thanks for the change. 👍

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/util/ParseUtils.java [hunk 1]
```diff
@@ -434,4 +436,9 @@ public static SearchSourceBuilder addUserBackendRolesFilter(User user, SearchSou
         }
         return searchSourceBuilder;
     }
+
+    public static User getUserContext(Client client) {
+        String userStr = client.threadPool().getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER_AND_ROLES);
+        return User.parse(userStr);
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/util/ParseUtils.java:442]: How about we add some debug log like the alerting PR https://github.com/opendistro-for-elasticsearch/alerting/pull/288/files#diff-a168ed1ef25a8c7d17602d9f10e5944d20280b0854364b4cf3498240921972cbR104?

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/util/ParseUtils.java:442]: Sure.

Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/util/ParseUtils.java:442]: Thanks, approve it, merge when you finish.

**Review by ylwu-amzn — APPROVED** [review]

**Review by yizheliu-amazon — APPROVED** [review]
thanks for the change and cleaner code.

**Pull request merged**